pkg/daemon/ceph/osd/device.go

```diff
@@ -77,6 +77,11 @@ type DeviceOsdIDEntry struct {
 	Metadata []int // OSD IDs (multiple) that have metadata stored here
 }
 
+func (m *DeviceOsdMapping) String() string {
+	b, _ := json.Marshal(m)
```

Comment:
is this just used for logging so we don't have to worry about the error?

Reply:
yes, all these `String()` functions are used for logging. I never expect an error from the marshal call and handling it does not seem appropriate.

The logging of all these types is so much richer now because the default `%+v` does not follow pointer addresses, but using a json marshal of the objects will give us all the fields (even in nested structs).